Packs/SafeBreach/Integrations/SafeBreach_v2/SafeBreach_v2.py

```diff
@@ -0,0 +1,887 @@
+    except TypeError:
+        demisto.info('Failed to sort SafeBreach insights, skip')
+
+    if insight_ids and len(insight_ids) > 0:
+        # Verify that insight_ids holds List[int]
+        if isinstance(insight_ids, list):
+            insight_ids = list(map(int, insight_ids))
+        insights = list([item for item in insights if int(item.get('ruleId')) in insight_ids])
+    insight_output = []
+    insight_readable = []
+    headers = []
+
+    for insight in insights:
+        context_insight = {
+            'Name': insight['actionBasedTitle'],
+            'Id': insight['ruleId'],
+            'DataType': INSIGHT_DATA_TYPE_MAPPER.get(insight['ruleId']) or 'Other',
+            'Category': insight.get('category'),
+            'LatestSimulation': insight.get('maxExecutionTime'),
+            'EarliestSimulation': insight.get('minExecutionTime'),
+            'SimulationsCount': insight.get('context').get('simulationsCount'),
+            'RiskImpact': float("{0:.2f}".format(insight.get('impact'))),
+            'AffectedTargetsCount': len(insight.get('targets')),
+            'SeverityScore': insight.get('severityScore'),
+            'Severity': insight.get('severity'),
+            'RemediationDataCount': insight.get("mitigationPoints").get('value'),
+            'RemediationDataType': insight.get("mitigationPoints").get('key'),
+            'ThreatGroups': insight.get('threatActors'),
+            'NetworkDirection': insight.get('direction'),
+            'AttacksCount': len(insight.get('attacks')),
+        }
+        t = {
+            'Id': context_insight['Id'],
+            'Name': context_insight['Name'],
+            'Category': context_insight['Category'],
+            'Risk Impact': context_insight['RiskImpact'],
+            'Severity': context_insight['Severity'],
+            'Affected Targets': context_insight['AffectedTargetsCount'],
+            'Data Type': context_insight['DataType'],
+        }
+        headers = list(t.keys())
+        insight_output.append(context_insight)
+        insight_readable.append(t)
+    readable_output = tableToMarkdown(name='SafeBreach Insights', t=insight_readable, headers=headers,
+                                      removeNull=True)
+    outputs = {
+        'SafeBreach.Insight(val.Id == obj.Id)': insight_output
+    }
+    if no_output_mode:
+        return_outputs(
+            readable_output,
+            outputs,
+            insight_output
+        )
+    return insights
+
+
+def get_test_status_command(client: Client, args: Dict):
+    """Get status of a SafeBreach test for tracking progress of a run.
+
+            Arguments:
+                client {Client} -- Client derives from BaseClient
+                args {dict}  -- function's arguments
+
+            Keyword Arguments:
+                    testId : float
+            Returns:
+               None
+            """
+    test_id = args.get('testId')
+    tries = 0
+    time.sleep(5)
```

Comment:
remove sleep,  sleep have big impact to the performance in demisto. instead for async actions we have GenericPolling playbooks.
https://xsoar.pan.dev/docs/playbooks/generic-polling

Reply:
We have added it because it takes time to create test at SafeBreach, As Francesco suggests we added the sleep within the command to avoid errors when users first call rerun and than get-status command.
Alternatively, we can remove it from the code and leave it to the user to handle the risk of calling the command before the test actually created. it will fail the command and failed the playbook that using this command.